### PR TITLE
chore: Remove raw SQL query

### PR DESCRIPTION
**Pull request by deepeshgarg007**
Removed raw SQL query and fixed typos

### CODE REVIEW

**Comment by deepeshgarg007** [comment]
@Mergifyio backport version-13-hotfix version-14-hotfix

File: erpnext/accounts/doctype/bank_guarantee/bank_guarantee.py [hunk 1]
```diff
@@ -25,14 +22,17 @@ def on_submit(self):
...
+
+	doctype = "Sales Order" if bank_guarantee_type == "Receiving" else "Purchase Order"
+
+	if doctype == "Sales Order":
+		fields_to_fetch.append("customer")
+		fields_to_fetch.append("project")
+	else:
+		fields_to_fetch.append("supplier")
+
+	bg_doctype = frappe.qb.DocType("Bank Guarantee")
```
Comment (sagarvora) [inline erpnext/accounts/doctype/bank_guarantee/bank_guarantee.py]: remove?

**Comment by sagarvora** [comment]
I've added a type check for `reference_name`.

**Comment by deepeshgarg007** [comment]
@Mergifyio backport version-14-hotfix version-13-hotfix

**Pull request merged**